osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmgmt/application/config/SecurityConfig.java

```diff
@@ -0,0 +1,54 @@
+package org.opensmartgridplatform.secretmgmt.application.config;
+
+import org.opensmartgridplatform.secretmgmt.application.services.encryption.DefaultEncryptionDelegate;
+import org.opensmartgridplatform.secretmgmt.application.services.encryption.EncryptionDelegate;
+import org.opensmartgridplatform.secretmgmt.application.services.encryption.providers.EncryptionProvider;
+import org.opensmartgridplatform.secretmgmt.application.services.encryption.providers.EncryptionProviderType;
+import org.opensmartgridplatform.secretmgmt.application.services.encryption.providers.JreEncryptionProvider;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.Resource;
+
+import java.util.Optional;
+
+@Configuration
+public class SecurityConfig {
+
+    @Value("${soap.secret.resource}")
+    private Resource soapSecretResource;
+
+    @Value("${database.secret.resource}")
+    private Resource databaseSecretResource;
+
+    @Value("${hsm.keystore.resource:#{null}}")
+    private Optional<Resource> hsmKeystoreResource;
+
+    @Value("${encryption.provider.type}")
+    private String encryptionProviderTypeName;
+
+    @Bean(name = "osgpEncryptionDelegate")
+    public EncryptionDelegate getDefaultEncryptionDelegate() throws Exception {
+        EncryptionDelegate encryptionDelegate;
```

Comment:
Note that the DefaultEncryptionDelegate is not a delegate but a Delegator. The actual Delegates are the providers doing the work.
Consider renaming.